### PR TITLE
stdlib and libm changes

### PR DESCRIPTION
**Pull request by mattuna15**
- add rand() function to stdlib
- add game of life example
- add libm (Maths Lib)
- add libm example

### CODE REVIEW

File: code/software/libm-test/kmain.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2020 You (you@youremail.com)
```
Comment (roscopeco) [inline code/software/libm-test/kmain.c:2]: Would you mind updating this with your copyright info please? :)

File: code/software/libm-test/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# C Starter Project for rosco_m68k
```
Comment (roscopeco) [inline code/software/libm-test/README.md:1]: Could we either update this file, or remove it please? :)

File: .gitignore [hunk 3]
```diff
@@ -22,3 +22,5 @@
 .vscode/*
 code/software/lcd-ili9341/images/BMPToRLE555
 code/software/lcd-ili9341/rle_images.h
+c_cpp_properties.json
```
Comment (roscopeco) [inline .gitignore:25]: Are these changes needed? They feel a bit editor-specific...?

File: code/software/libs/src/cstdlib/include/stdlib.h [hunk 4]
```diff
@@ -22,12 +22,15 @@
...
 
+#define RAND_MAX 32767
+
 void exit(int status);
 void abort(void);
 void *malloc(size_t size);
 void free(void *ptr);
 void *realloc(void *ptr, size_t size);
 void *calloc(size_t nitems, size_t size);
+int rand(void);
```
Comment (roscopeco) [inline code/software/libs/src/cstdlib/include/stdlib.h:33]: Is this still available if `libm` isn't linked?

Comment (roscopeco) [reply code/software/libs/src/cstdlib/include/stdlib.h:33]: (Just saw following file, ignore this comment :) )

File: code/software/libs/src/libm/common/Makefile.am [hunk 5]
```diff
@@ -0,0 +1,76 @@
+## Process this file with automake to generate Makefile.in
```
Comment (roscopeco) [inline code/software/libs/src/libm/common/Makefile.am:1]: Does this still work? If not, I'd prefer we omit the automake files and just commit the generated Makefile...

File: code/software/libs/src/libm/common/acosl.c [hunk 6]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (roscopeco) [inline code/software/libs/src/libm/common/acosl.c:1]: (Caveat: I've not checked the whole PR): I would like to add this licence text at the top level, along with the other licences (perhaps as `LICENCE.libm` or something).

Comment (roscopeco) [reply code/software/libs/src/libm/common/acosl.c:1]: (Also, gonna take the rest of the PR as read, given that I think it's all IBM code? Please correct me if I'm wrong...)

Comment (roscopeco) [reply code/software/libs/src/libm/common/acosl.c:1]: (Okay, I see there are numerous licences in here, IBM, ARM, Sun, RedHat. Nothing worrying, but If we could reproduce those licenses at the top level that would be super helpful in terms of reproducing them :) )

**Review by roscopeco — CHANGES_REQUESTED** [review]

File: code/software/libs/src/libm/common/acosl.c (hunk 6 above)
Comment (mattuna15) [reply code/software/libs/src/libm/common/acosl.c:1]: These are all the companies who've contributed over time to libm. Amazing really, when you think about it. I'll copy them. I've included the newlib licenses.

**Review by roscopeco — APPROVED** [review]
Looks good, thanks again @mattuna15  :)
